lib/syskit/cli/log_runtime_archive.rb

```diff
@@ -185,7 +185,23 @@ def self.transfer_dataset(
                 result = TransferDatasetResult.new(
                     complete: complete, transfer_results: transfer_results
                 )
-                log_transfer_results(dataset_path, result, logger: logger)
+                results = log_transfer_results(dataset_path, result, logger: logger)
+                remove_dataset_folder_if_empty(dataset_path, logger)
+                results
+            end
+
+            def self.remove_dataset_folder_if_empty(dataset_path, logger)
```

Comment:
Reminder: change this logic to delete the folder if it is not locked and the only file present is `.lock`